client/src/resources/ItemStore.js

```diff
@@ -133,68 +133,40 @@ export default class ItemStore {
     return this.items[id];
   }
 
-  async getSearchRequestBody(searchString, limit, onlyTools) {
-    let queries = [];
+  async getSearchUrl(searchString, limit, bookmark) {
+    const QServerBaseUrl = await qEnv.QServerBaseUrl;
+    let searchUrl = `${QServerBaseUrl}/search?`;
     for (let filter of this.filters) {
       if (filter.name === "tool" && filter.selected !== "all") {
-        queries.push(
-          `(tool:"${filter.selected}" OR tool:${filter.selected.replace(
-            new RegExp("_", "g"),
-            "-"
-          )})`
-        );
-
-        // we do have a specific tool filter, so we do not want to have the onlyTools processed
-        onlyTools = null;
+        searchUrl = searchUrl + "tool=" + filter.selected + "&";
```

Comment:
Please use https://nodejs.org/api/querystring.html, construct an object of all query parameters and then `querystring.stringify` in the end.

Reply:
Good idea. I have implemented this using the URLSearchParams web api as this is on client side -> https://github.com/nzzdev/Q-editor/pull/97/commits/8f15c0039e09d0f58bda3a31a47b69fc2b4f7d30